### ISSUE TITLE
Consider rendering indoor=room + shop/amenity better

### ISSUE THREAD

**Issue by CloCkWeRX** [gravitystorm/openstreetmap-carto#2417]
OpenLevelUp does a good job of rendering shopping malls and similar; but the default openstreetmap style doesn't do as well.

For example:
http://www.openstreetmap.org/#map=18/-27.38372/153.03169

Zoom 19, openstreetmap-carto
![image](https://cloud.githubusercontent.com/assets/365751/19623903/ab7adf8e-9926-11e6-88d0-670b5e2cb364.png)
 * Good detail
 * Its not really clear how the shops are accessed, how big they are, etc

High zoom, openlevelup
![image](https://cloud.githubusercontent.com/assets/365751/19623899/971f507e-9926-11e6-980f-5f863eb1908d.png)
 * Gives some idea of layout of shops
 * Better for navigating internally

Non free map; high zoom:
![image](https://cloud.githubusercontent.com/assets/365751/19623913/d88f7764-9926-11e6-8f23-db86677257cd.png)

Zoom level 17:
![image](https://cloud.githubusercontent.com/assets/365751/19623931/b931092c-9927-11e6-9c91-b60c89380017.png)
 * Labels are dropped, but the food court is still very visually noisy

Zoom level 16:
![image](https://cloud.githubusercontent.com/assets/365751/19623939/f727e3cc-9927-11e6-9afe-104c0f984bda.png)
 * Much cleaner visually

I'd be interesting in seeing what zoom level 17, indoor=* + amenity=fast_food or shop=* looked like if icons were replaced with the same "dot" for shop=yes; or a similar smaller icon:

Mockup:
![image](https://cloud.githubusercontent.com/assets/365751/19623973/a3ff31e4-9929-11e6-9386-00bd14321ed6.png)

At zoom level 18-19, it seems like a slightly lighter background color on indoor=* could be beneficial.

Mockup, please ignore the blue border/yellow boxes from JOSM:
![image](https://cloud.githubusercontent.com/assets/365751/19623981/0654e262-992a-11e6-8a6e-45019183f07b.png)

**Comment by mboeringa** [comment]
I think it very unwise to start rendering indoor room features in the main style. OpenLevelUp was specially designed for this, and is able to show features **_per level_**. In a pre-rendered style like Carto, you can not show multi-level building layout properly, everything will be stacked on top of each other.

While this particular single level shopping mall example might work reasonably in the mock-up, multi-level shopping malls with differing layout on levels, will become a mess in a pre-rendered style like Carto.

**Comment by matkoniecz** [comment]
Yes, it is rather typical that multilevel malls will have different syore layouts on each level. So rendering shop areas is unlikely to work well.

**Comment by CloCkWeRX** [comment]
>  everything will be stacked on top of each other.

What about the zoom level 16 behaviour that is currently there - thats already somewhat of a problem with the icons etc. What can be done to reduce that clutter?

**Comment by mboeringa** [comment]
> What about the zoom level 16 behaviour that is currently there - thats already somewhat of a problem with the icons etc. What can be done to reduce that clutter?

My personal subjective opinion? ;-)

That would be that all shops belong at Z19/20... and even there they often collide with addresses, making the rendering of both impossible, so both rendering suffer: you neither see all addresses, nor all shops.

Even better would be a dynamic Overpass overlay that could be switched on / off at will, instead of as part of the rendering. But that would require changes to the main OpenStreetMap website...

**Comment by dieterdreist** [comment]
I also agree that the cases where indoor rendering results in a better
general map are likely much fewer than those where it will lead to
confusion and inferiour rendering quality. I could imagine rendering the
"default" floor (i.e. ground floor, sometimes 1st or 1st basement floor
according to local numbering) of indoor spaces could improve generally the
map at high zoom levels. For reference, Giambattista Nolli did exactly this
in his world famous Rome city map from 1748 ( http://nolli.uoregon.edu/
http://nolli-app.com ), and is still famous and referenced a lot for his
equal treatment of public space regardless of outdoor or indoor location
(BUT: we in OSM don't have an established way to distinguish indoor spaces
between semi-public and private).

**Comment by kocio-pl** [comment]
How should we select the default floor? I guess we can just avoid rendering shop/amenity with level tag present and different than 0. 

However I'm not sure about side effects. Amenities and shops with level=* tag may be located in the underground passages (do we want them to still be visible or not?) and possibly in other places, not only in multi-level malls.

**Comment by matkoniecz** [comment]
I recommend closing this issue as it is unable to support malls with more than a single floor.

Showing only level=0 pois would break rendering in significant number of cases (underground passages, restaurants in basements, multilevel malls and as result encourage tagging for rendering by removing level tag) and would be quite strange - why show object at ground floor and not on upper floors?
